pyatv/__init__.py

```diff
@@ -41,10 +52,13 @@ def _should_include(atv):
         return True
 
     scanner: BaseScanner
-    if hosts:
-        scanner = UnicastMdnsScanner([IPv4Address(host) for host in hosts], loop)
+    if async_zeroconf:
```

Comment:
Do you see any benefits of keeping any of the old implementations? I would prefer if we could just remove them altogether and just use your new one instead. It would remove so much code and so many obnoxious tests...

Reply:
A service browser has to be running for the types for the zeroconf implementation to work. 

Its easy enough to start one and then kill it off at the end of the scan, but I didn't want to bite off too much in a single PR.

Reply:
Also I didn't implement knocking in the zc scanner.  Everything seemed to work fine without it.  If its needed, I'll do another turn with it later.

Reply:
Yeah, I expect that we create one internally and close it when we are done. It's the best compromise I would say.

Port knocking is only relevant for devices that goes to deep sleep. I have only encountered that with my Apple TV 3, where the port knocking feature is needed.

Reply:
And also, since the changes you are making does not alter any public interface we can let mine and your implementation live side by side and then we just remove my stuff after merging your changes. Just to keep it clean-ish.

Reply:
> Port knocking is only relevant for devices that goes to deep sleep. I have only encountered that with my Apple TV 3, where the port knocking feature is needed.

Do you still have one of these for testing? I know they are EOL but it would be nice to test with it if we can.